src/Squidex.Domain.Apps.Entities.MongoDb/Assets/MongoAssetRepository.cs

```diff
@@ -34,44 +41,40 @@ protected override Task SetupCollectionAsync(IMongoCollection<MongoAssetEntity>
         {
             return collection.Indexes.CreateOneAsync(
                 Index
-                    .Ascending(x => x.State.AppId)
-                    .Ascending(x => x.State.IsDeleted)
-                    .Ascending(x => x.State.FileName)
-                    .Ascending(x => x.State.MimeType)
-                    .Descending(x => x.State.LastModified));
+                    .Ascending(x => x.AppId)
+                    .Ascending(x => x.IsDeleted)
+                    .Ascending(x => x.FileName)
+                    .Ascending(x => x.MimeType)
+                    .Descending(x => x.LastModified));
         }
 
-        public async Task<IResultList<IAssetEntity>> QueryAsync(Guid appId, HashSet<string> mimeTypes = null, HashSet<Guid> ids = null, string query = null, int take = 10, int skip = 0)
+        public async Task<IResultList<IAssetEntity>> QueryAsync(Guid appId, string query = null)
```

Comment:
For IContentRepository we pass in the ODataUriParser. We should be consistent here.

Reply:
IContentRepository has the ContentQueryService over it, which is where converting string to ODataUriParser exits.  IAssetRepository doesn't have that abstraction.  I will add an AssetQueryService to do this.

Reply:
You are right, then just keep it like that.